sr/comp/cli/show_match_scores.py

```diff
@@ -0,0 +1,167 @@
+__description__ = "Show the game and league points achieved for each match"
+
+from typing import Dict, List, Union, NamedTuple
+from sr.comp.types import TLA, GamePoints, ArenaName, MatchNumber
+from sr.comp.scores import LeaguePosition
+
+
+class MatchCorner(NamedTuple):
+    tla: TLA
+    ranking: Union[LeaguePosition, str]  # to allow "???" to be used for unknown scores
+    game: Union[GamePoints, str]
+    league: Union[int, str]
+
+
+class MatchResult(NamedTuple):
+    num: MatchNumber
+    arena: ArenaName
+    display_name: str
+    corners: Dict[int, MatchCorner]
+
+
+def collect_match_info(comp, match) -> MatchResult:
+    from sr.comp.match_period import MatchType
+    from sr.comp.scores import degroup
+
+    if match.type == MatchType.knockout:
+        score_data = comp.scores.knockout
+    elif match.type == MatchType.tiebreaker:
+        score_data = comp.scores.tiebreaker
+    elif match.type == MatchType.league:
+        score_data = comp.scores.league
+
+    match_id = (match.arena, match.num)
+
+    if match_id in score_data.game_points:
+        league_points = score_data.ranked_points[match_id]
+        game_points = score_data.game_points[match_id]
+        ranking = degroup(score_data.game_positions[match_id])
+    else:
+        league_points = {}
+        game_points = {}
+        ranking = {}
+        for team in match.teams:
+            league_points[team] = "???"
+            game_points[team] = "???"
+            ranking[team] = "???"
+
+    corner_data: Dict[int, MatchCorner] = {}
+
+    for corner, team in enumerate(match.teams):
+        match_id = (match.arena, match.num)
+
+        if team:  # corner occupied
+            corner_data[corner] = MatchCorner(
+                tla=team,
+                ranking=ranking[team],
+                game=game_points[team],
+                league=league_points[team],
+            )
+        else:
+            corner_data[corner] = MatchCorner(
+                tla='',
+                ranking='',
+                game='',
+                league='',
+            )
+
+    return MatchResult(
+        num=match.num,
+        arena=match.arena,
+        display_name=match.display_name,
+        corners=corner_data,
+    )
+
+
+def print_col(text):
+        print(text, end='|')
+
+
+def print_heading(num_corners, name_width, arena_name_width):
+    print_col("".center(name_width + 1 + arena_name_width))
+    for idx in range(num_corners):
+        print_col(f"Zone {idx}".center(22))
```

Comment:
Please can we remove this magic number? (what does 22 mean here?) Ditto for the 5 in `"TLA".center(5)` below.

Reply:
These have been removed now we're using a library for drawing the table. 8bde289db696a2afd6ca97141f3feecf352ec890